articles/adding-dark-theme-to-your-site/index.md

```diff
@@ -0,0 +1,178 @@
+        -webkit-transition: 0.3s;
+        transition: 0.3s;
+    }
+
+    input:checked + label {
+        background: var(--color-heading);
+    }
+
+    input:checked + label:after {
+        left: calc(100% - 5px);
+        -webkit-transform: translateX(-100%);
+                transform: translateX(-100%);
+    }
+
+    label:active:after {
+        width: 45px;
+    }
+```
+
+### Adding the JavaScript
+We will handle the theme switching through JavaScript by changing the toggle switch's class name to either light or dark, as defined above. In doing so, we write two functions `changeThemeToDark()` and `chnageThemeToWhite()`.
+
+**Change the theme to dark.**
+```javascript
+    // Change theme to dark by adding the `dark` classname to html element.
+   const changeThemeToDark = () =>{
+    document.documentElement.setAttribute("data-theme", "dark")
+}
+
+```
+
+**Change the theme to Light.**
+```javascript
+    // Reset the html class to default
+   const changeThemeToDark = () =>{
+    document.documentElement.setAttribute("data-theme", "dark");
+}
+```
+
+### Using Local Storage to remember the theme.
+Now that our dark theme is set, we need to enable the browser to remember the theme our user chose so that in subsequent visits or page refreshing, they see their preferred theme. We can use local storage to implement this functionality. We will store the applied theme in local storage and read subsequent calls of the `changeThemeToDark()` and `chnageThemeToWhite().` Even after refreshing the page, the theme is still reserved. We are using local storage because the data stored in local storage has no expiration time. The 
+The below pieces of code are used to save and retrieve the theme from local storage:
+```javascript
+    const changeThemeToDark = () =>{
+        document.documentElement.setAttribute("data-theme", "dark")
+        localStorage.setItem("data-theme", "dark")// save theme to local storage
+    }
+
+    const changeThemeToLight = () =>{
+        document.documentElement.setAttribute("data-theme", "light")
+        localStorage.setItem("data-theme", 'light')// save theme to local storage
+    }
+```
+After writing, the functions, we check to see what theme is set currently and toggle it.
+```javascript
+    // Apply retrived them to the website
+    checkbox.addEventListener('change', ()=> {
+        let theme = localStorage.getItem('data-theme'); // Retrieve saved them from local storage
+        if (theme ==='dark'){
+            changeThemeToLight()
+        }else{
+            changeThemeToDark()
+        }   
+    });
+```
+### Conclusion
+To sum up, since dark themes are trendy recently, this article explained how to implement dark themes and light themes switching using CSS variables and Javascript's LocalStorage. You can now try out the dark theme on your website following this article. You can find the link to the code used in the article [here](https://replit.com/@PhinaKersly/dark-theme#index.html)
```

Comment:
```suggestion
To sum up, since dark themes are trendy recently, this article explained how to implement dark themes and light themes switching using CSS variables and Javascript's localStorage.

You can now try out the dark theme on your website following this article. You can find the link to the code used in the article [here](https://replit.com/@PhinaKersly/dark-theme#index.html)
```